lib/util/string_utils.js

```diff
@@ -329,6 +329,28 @@ shaka.util.StringUtils = class {
     }
     return input || '';
   }
+
+  /**
+   * Calculate the hash of an input
```

Comment:
This function is impossible to review.

Reply:
:( Why?

Reply:
This is not the kind of math I can verify in my head.  🙃

Seriously, though, if we're relying on this hash to generate IDs and avoid conflicts, I'm not sure a random snippet from stack overflow is a good idea.  It would be better if:

1. it were provided by the browser environment or a known-trustworthy JS implementation
2. the hash size were more than 32 bits.

Reply:
I will look for an alternative....